pdg/src/info.rs

```diff
@@ -67,18 +67,483 @@ fn set_flow_info(g: &mut Graph) {
         }
         node.info = Some(NodeInfo {
             flows_to: cur_node_flow_info,
+            unique: false,
         });
     }
 }
 
+/// Gathers information from a [`Graph`] (assumed to be acyclic and topologically sorted but not
+/// necessarily connected) for each [`Node`] in it what its chronologically (judged by [`NodeId`])
+/// final descendent is.
+fn get_last_desc(g: &mut Graph) -> HashMap<NodeId, NodeId> {
+    let mut desc_map: HashMap<NodeId, NodeId> =
+        HashMap::from_iter(g.nodes.iter_enumerated().map(|(idx, _)| (idx, idx)));
+    for (n_id, node) in g.nodes.iter_enumerated().rev() {
+        if let Some(p_id) = node.source {
+            let cur_node_last_desc: NodeId = *desc_map.get(&n_id).unwrap();
+            let parent_last_desc: NodeId = desc_map.remove(&p_id).unwrap();
+            desc_map.insert(p_id, std::cmp::max(cur_node_last_desc, parent_last_desc));
+        }
+    }
+    desc_map
+}
+
+/// Finds the inverse of a [`Graph`], each [`Node`] mapping to a list of its children.
+fn collect_children(g: &Graph) -> HashMap<NodeId, Vec<NodeId>> {
+    let mut m = HashMap::new();
+    for par in g.nodes.indices() {
```

Comment:
What's `par`?

Reply:
each possible parent

Reply:
`par` is confusing.  Can we just call it `parent`?  `par` is more often `parallel`.  Same with `chi` and `child`.